qcodes/instrument_drivers/Lakeshore/Model_325.py

```diff
@@ -278,41 +276,14 @@ def __init__(self, parent: 'Model_325', name: str, inp: str) -> None:
             vals=Numbers(min_value=1, max_value=35)
         )
 
-    def decode_sensor_status(self, sum_of_codes: int) -> str:
-        """
-        The sensor status is one of the status codes, or a sum thereof. Multiple
-        status are possible as they are not necessarily mutually exclusive.
-
-        args:
-            sum_of_codes (int)
-        """
-        components = list(self.sensor_status_codes.keys())
-        codes = self._get_sum_terms(components, int(sum_of_codes))
-        return ", ".join([self.sensor_status_codes[k] for k in codes])
-
     @staticmethod
-    def _get_sum_terms(components: list, number: int):
-        """
-        Example:
-        >>> components = [0, 1, 16, 32, 64, 128]
-        >>> _get_sum_terms(components, 96)
-        >>> ...[64, 32]  # This is correct because 96=64+32
-        """
-        if number in components:
-            terms = [number]
-        else:
-            terms = []
-            comp = np.sort(components)[::-1]
-            comp = comp[comp <= number]
-
-            while len(comp):
-                c = comp[0]
-                number -= c
-                terms.append(c)
-
-                comp = comp[comp <= number]
-
-        return terms
+    def decode_sensor_status(sum_of_codes: int, status=Status) -> str:
+        total_status = status(sum_of_codes)
+        if sum_of_codes == 0:
+            return 'OK'
+        status_messages = [st.name.replace('_', ' ') for st in status
+                           if st in total_status]
+        return ", ".join(status_messages)
```

Comment:
Why is this `, status=Status` argument needed? the static method can access the "class attribute `Status`" without any problems. Also, any external code can access the "class attrbiute `Status`" via the class (not instance): `Model_325_Sensor.Status` - that's it :)

i leave it up to you, but i think it's also fine to declare the `Status` of the module level, not necessarily inside the class.

Reply:
I have made it module level.